src/components/autocomplete/autocomplete.stories.js

```diff
@@ -0,0 +1,140 @@
+
+//--------------------------
+
+export const WithCustomInputField = () => html`
+  <vl-autocomplete data-min-chars="1" 
+                   data-static-list='[
+              {"title":"hello", "value": "1"},
+              {"title":"hi", "value": "2"},
+              {"title":"How are you", "value": "3" },
+              {"title":"My name is" , "value": "4"}]'>
+    <input type="text" placeholder="Hint: typ hello, hi, How are you of My name is" style="width: 100%" />
+  </vl-autocomplete>`;
+
+//--------------------------
+
+async function fetchDataFromFunction(searchTerm, autocomplete){
+  const suggestions = [
+    {"title":"hello", "value": "1"},
+    {"title":"hi", "value": "2"},
+    {"title":"How are you", "value": "3" },
+    {"title":"My name is" , "value": "4"}];
+  autocomplete.filterAndSuggest(searchTerm, suggestions);
+}
+
+export const WithDataFromFunction = () => html`
+  <vl-autocomplete data-min-chars="1" .dataFetcher="${fetchDataFromFunction}" placeholder="Hint: typ hello, hi, How are you of My name is"></vl-autocomplete>`;
+
+//--------------------------
+
+async function fetchComplexDataFromFunction(searchTerm, autocomplete){
+  const suggestions = [
+    {"title":"Gent", "subtitle":"Gemeente", "value": "1"},
+    {"title":"Gentbos, Merelbeke", "subtitle":"Adres", "value": "2"},
+    {"title":"Gentbruggestraat, Gent", "subtitle":"Adres", "value": "3"},
+    {"title":"Gentele, Brugge", "subtitle":"Adres", "value": "5"},
+    {"title":"Automotive Contractors Gent ", "subtitle":"Project", "value": "6"},
+    {"title":"Buurtshuis Watersportbaan Gent", "subtitle":"Project", "value": "7"}]
+  autocomplete.filterAndSuggest(searchTerm, suggestions);
+}
+
+export const WithComplexDataFromFunctionAndDefaultCaptionFormatter = () => html`
+  <vl-autocomplete data-min-chars="1" .dataFetcher="${fetchComplexDataFromFunction}" placeholder="Hint: typ Gent"></vl-autocomplete>`;
+
+//--------------------------
+function formatCaptionWithOnlyTitle(item)
+{
+  return item.title;
+}
+
+
+export const WithComplexDataFromFunctionGroupBySubTitle = () => html`
+  <vl-autocomplete data-min-chars="1" data-group-by="subtitle" .dataFetcher="${fetchComplexDataFromFunction}" 
+                   .captionFormatter="${formatCaptionWithOnlyTitle}" 
+                   placeholder="Hint: typ Gent"></vl-autocomplete>`;
+
+//--------------------------
+
+function formatCaption(item)
+{
+  if(item.subtitle != null) {
+    return `${item.subtitle}: ${item.title}`;
+  }
+  else {
+    return item.title;
+  }
+}
+
+export const WithComplexDataFromFunctionAndCustomCaptionFormatter = () => html`
+  <vl-autocomplete data-min-chars="1" .dataFetcher="${fetchComplexDataFromFunction}" .captionFormatter="${formatCaption}" placeholder="Hint: typ Gent"></vl-autocomplete>`;
+
+//--------------------------
+
+function formatCaptionAsHtml(item)
+{
+  return `${item.subtitle}<br>${item.title}`;
+  //return html`${item.subtitle}<br>${item.title}`;
```

Comment:
Normaal is het niet de bedoeling dat we commented code  gaan mee committen.

Reply:
Die comment zou inderdaad uiteindelijk weg moeten maar die staat er nog omdat dat verschillende probeersels zijn om de tekst die getoond wordt per item in de suggestie lijst te formatteren met HTML. Daar zit ik nog met een probleem (zoals omschreven in deze pull request). Ik zou die dus nog even laten staat tot dit probleem is opgelost.